### PR TITLE
Add firebase datastore

### PR DESCRIPTION
**Pull request by sburba**

### CODE REVIEW

**Comment by sburba** [comment]
This is based on the zeit PR, so that should be reviewed first

**Comment by sburba** [comment]
Oh, speaking of which, I need to set up the firebase creds in the now.json

**Comment by sburba** [comment]
> Oh, speaking of which, I need to set up the firebase creds in the now.json, so don't merge this until I do that

Okay, that was more difficult than I expected, but it's once again possible to deploy this to zeit.

**Comment by sburba** [comment]
> Need to wait for [zeit/webpack-asset-relocator-loader#12](https://github.com/zeit/webpack-asset-relocator-loader/pull/12) to actually get deployed for this to start working in zeit :(

This actually wasn't the problem, the problem was needing all the types in the production dependencies block in the package.json

File: server/firestore/firestore.indexes.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  // Example:
```
Comment (mawills) [inline server/firestore/firestore.indexes.json]: I don't think you can have comments in a .json file

Comment (sburba) [reply server/firestore/firestore.indexes.json]: Definitely can't. Although the firebase json pre-processor allows it. I can pull them out if you prefer

**Pull request merged**